src/ensembl/src/content/app/entity-viewer/components/transcript-filter/EntityViewerTranscriptFilter.test.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import times from 'lodash/times';
+import faker from 'faker';
+
+import EntityViewerTranscriptFilter from './EntityViewerTranscriptFilter';
+import Checkbox from 'src/shared/components/checkbox/Checkbox';
+
+let totalOptions = 0;
+
+const createOption = () => {
+  totalOptions++;
+  const randomValueAndLabel = faker.random.uuid();
+
+  return {
+    value: randomValueAndLabel,
+    label: randomValueAndLabel
+  };
+};
+
+const createOptionGroup = (number = 5) => {
+  return times(number, () => createOption());
+};
+
+const onChange = jest.fn();
+
+const optionGroups = times(5, () => createOptionGroup());
+
+const getRandomSelectedValues = () => {
+  const randomSelectedValues: string[] = [];
+
+  optionGroups.forEach((options) =>
+    randomSelectedValues.push(
+      options[Math.floor(Math.random() * options.length)].value
+    )
+  );
+
+  return randomSelectedValues;
+};
+
+const selectedValues = getRandomSelectedValues();
+
+const defaultProps = {
+  optionGroups: optionGroups,
+  selectedValues: selectedValues,
+  onChange
+};
+
+const renderTranscriptFilter = (props?: any) => {
```

Comment:
Surely not `any`. Looks like it's `Partial<EntityViewerTranscriptFilterProps>`.

Reply:
That's done